### PR TITLE
[HPRO-1257] Implement sample and aliquot finalization

### PR DESCRIPTION
**Pull request by shyamzcode**
| Q                   | A
| ------------------- | --------------
| Target Release?     | 3.0.0 
| Bug fix?            | ❌ 
| New feature?        | ✅ 
| Database migration? | ❌ 
| New configuration?  | ❌ 
| Composer updates?   | ❌ 
| NPM updates?        | ❌ 
| Jira Ticket(s)      | HPRO-1257 

### Screenshots 

image

### CODE REVIEW

File: src/Service/Nph/NphOrderService.php [hunk 1]
```diff
@@ -372,4 +378,65 @@ public function getParticipantOrderSummaryByModuleAndVisit(string $participantid
         }
         return $returnArray;
     }
+
+    public function saveOrderFinalization($formData, $sample)
```
Comment (thornview) [inline src/Service/Nph/NphOrderService.php:382]: I noticed that most of the parameters in this service are not type-hinted.  Can you add type-hints for these parameters?

File: src/Form/Nph/NphOrderForm.php [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        'I had normal formed stool, and my stool looks like Type 3 and/or 4' => 'normal'
+    ];
+
+    public static $bowelMovementQuality = [
+        'I tend to be constipated (have difficulty passing stool) - Type 1 and 2' => 'difficult',
+        'I tend to have diarrhea (watery stool) - Type 5, 6, and 7' => 'watery',
+        'I tend to have normal formed stool - Type 3 and 4' => 'normal'
+    ];
+
+    protected function addCollectedSampleFields($builder, $sample, $sampleLabel): void
```
Comment (thornview) [inline src/Form/Nph/NphOrderForm.php:41]: Please add type-hints to the parameters for the methods in this class.

File: src/Service/Nph/NphOrderService.php [hunk 3]
```diff
@@ -372,4 +378,65 @@ public function getParticipantOrderSummaryByModuleAndVisit(string $participantid
...
+            $sample->setSampleMetadata($this->jsonEncodeMetadata($formData, ['urineColor',
+                'urineClarity']));
+        }
+        $this->em->persist($sample);
+        $this->em->flush();
+        $this->loggerService->log(Log::NPH_SAMPLE_UPDATE, $sample->getId());
+        return $sample;
+    }
+
+    public function getExistingSampleData($sample): array
```
[the remark this answers is not in the document]
Comment (shyamzcode) [reply src/Service/Nph/NphOrderService.php:417]: I covered this within the [testSaveOrderFinalization](https://github.com/all-of-us/healthpro/pull/1244/commits/029ddc012ab3df631a84ad7dad11b94a54f2ecb2#diff-c2d1d6ed7d573125db63fb639607e6f0055b905d89e0068b8c684c5fd601c902R257) method. Here I am first saving the finalize data and then checking if I am retrieving the exact same data.

File: src/Service/Nph/NphOrderService.php (hunk 1 above)
Comment (shyamzcode) [reply src/Service/Nph/NphOrderService.php]: Done!

File: src/Form/Nph/NphOrderForm.php (hunk 2 above)
Comment (shyamzcode) [reply src/Form/Nph/NphOrderForm.php]: Done!

**Review by Ehaic — APPROVED** [review]

**Pull request merged**